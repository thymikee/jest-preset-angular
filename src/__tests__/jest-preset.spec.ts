// @ts-expect-error for now we don't ship assets with .d.ts
import defaultsESMPreset from '../../presets/defaults-esm/jest-preset';
// @ts-expect-error for now we don't ship assets with .d.ts
import defaultsPreset from '../../presets/defaults/jest-preset';

describe('Jest presets', () => {
  test.each([defaultsPreset, defaultsESMPreset])('should return the correct jest config', (preset) => {
    expect(preset).toMatchSnapshot();
  });
});
