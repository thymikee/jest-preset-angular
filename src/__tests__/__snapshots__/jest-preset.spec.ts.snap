// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Jest presets should return the correct jest config 1`] = `
Object {
  "globals": Object {
    "ts-jest": Object {
      "stringifyContentPathRegex": "\\\\.html$",
      "tsconfig": "<rootDir>/tsconfig.spec.json",
    },
  },
  "moduleFileExtensions": Array [
    "ts",
    "html",
    "js",
    "json",
  ],
  "moduleNameMapper": Object {
    "^app/(.*)$": "<rootDir>/src/app/$1",
    "^assets/(.*)$": "<rootDir>/src/assets/$1",
    "^environments/(.*)$": "<rootDir>/src/environments/$1",
    "^src/(.*)$": "<rootDir>/src/$1",
  },
  "snapshotSerializers": Array [
    "jest-preset-angular/build/serializers/html-comment",
    "jest-preset-angular/build/serializers/ng-snapshot",
    "jest-preset-angular/build/serializers/no-ng-attributes",
  ],
  "testEnvironment": "jsdom",
  "transform": Object {
    "^.+\\\\.(ts|js|html)$": "jest-preset-angular",
  },
}
`;

exports[`Jest presets should return the correct jest config 2`] = `
Object {
  "extensionsToTreatAsEsm": Array [
    ".ts",
  ],
  "globals": Object {
    "ts-jest": Object {
      "stringifyContentPathRegex": "\\\\.html$",
      "tsconfig": "<rootDir>/tsconfig.spec.json",
      "useESM": true,
    },
  },
  "moduleFileExtensions": Array [
    "ts",
    "html",
    "js",
    "json",
  ],
  "moduleNameMapper": Object {
    "^app/(.*)$": "<rootDir>/src/app/$1",
    "^assets/(.*)$": "<rootDir>/src/assets/$1",
    "^environments/(.*)$": "<rootDir>/src/environments/$1",
    "^src/(.*)$": "<rootDir>/src/$1",
    "tslib": "<rootDir>/node_modules/tslib/tslib.es6.js",
  },
  "snapshotSerializers": Array [
    "jest-preset-angular/build/serializers/html-comment",
    "jest-preset-angular/build/serializers/ng-snapshot",
    "jest-preset-angular/build/serializers/no-ng-attributes",
  ],
  "testEnvironment": "jsdom",
  "transform": Object {
    "^.+\\\\.(ts|js|html)$": "jest-preset-angular",
  },
  "transformIgnorePatterns": Array [
    "node_modules/(?!tslib)",
  ],
}
`;
